src/components/my-widgets-score-semester-storage.jsx

```diff
@@ -1,66 +1,212 @@
+		const _selectedTable = state.storageData?.slice(startIndex, endIndex)
+		setState({...state, rowsPerPage: newValue, selectedTable: _selectedTable, startIndex: startIndex, endIndex: endIndex, pageNumber: 0, pages: pagesArr})
+	}, [state])
+
+	const onChangeTable = useCallback(_selectedTableName => {
+		const numPages = Math.ceil(state.storageData?.length/state.rowsPerPage)
+		const pagesArr = Array.from({length: numPages}, (_, i) => i + 1) // Generates list of ints 1 to numPages
+		const startIndex = 0
+		const endIndex = Math.min (state.storageData?.length, state.rowsPerPage + 1)
+		const _selectedTable = state.storageData?.slice(startIndex, endIndex)
+		const _isTruncated = _selectedTable.length > MAX_ROWS
+		setState({...state, selectedTable: _selectedTable, selectedTableName: _selectedTableName, isTruncated: _isTruncated, startIndex: startIndex, endIndex: endIndex, pageNumber: 0, pages: pagesArr})
+	}, [state])
+
+	const onChangePageNumber = useCallback(newPageNum => {
+		const startIndex = Math.min(state.storageData?.length, newPageNum*state.rowsPerPage)
+		const endIndex = Math.min (state.storageData?.length, (newPageNum + 1)*state.rowsPerPage)
+		const _selectedTable = state.storageData?.slice(startIndex, endIndex)
+		setState({...state, startIndex: startIndex, endIndex: endIndex, selectedTable: _selectedTable, pageNumber: newPageNum})
+	}, [state])
+
+	const getPaginateBtns = useCallback(() => {
+		let pagesArr = state.pages?.map((val, index) => {
+			return(<a className={`paginate_button ${index === state.pageNumber ? 'current' : ''}`}
+				aria-controls="DataTables_Table_1"
+				tabIndex="0"
+				onClick={() => {onChangePageNumber(index)}}
+				key={index}>
+				{index+1}
+			</a>)
+		})
+
+		// Compresses the buttons
+		if (pagesArr.length > 7) {
+			const paginateIconL = <span className="ellipsis" key={999}>…</span>
+			const paginateIconR = <span className="ellipsis" key={888}>…</span>
+			const curPage = state.pageNumber + 1
+			const numPages = pagesArr.length
+			let paginateType = numPages - curPage >= 4 ?  
+				(curPage <= 4 ? 
+					"left" : 
+					"middle") :
+				"right"
+
+			switch(paginateType) {
+				case ("right"):
+					pagesArr = [pagesArr[0]].concat(paginateIconR).concat(pagesArr.slice(pagesArr.length - 5, pagesArr.length))
+					break
+				case ("middle"):
+					pagesArr = [pagesArr[0]].concat(paginateIconL).concat(pagesArr.slice(curPage - 2, curPage + 1)).concat(paginateIconR).concat(pagesArr[pagesArr.length - 1])
+					break
+				case ("left"):
+					pagesArr = pagesArr.slice(0, 5).concat(paginateIconL).concat(pagesArr[pagesArr.length - 1])
+					break
+			}
+		}
+
+		setPageinateBtns(pagesArr)
+	}, [state, searchInput])
+
+	// Filters search
+	const getFilter = useCallback((val) => {
+		const firstLast = val.play.firstName + val.play.lastName
+		const sanatizedSearch = searchInput.replace(/\s+/g, '').toUpperCase()
+
+		if (searchInput.length === 0)
+			return true
```

Comment:
avoid putting an `if` w/o curly brackets on 2 lines like the plague. Either a single line or use curly brackets.

Reason: it's just too easy for future devs to do something unintentionally here
```suggestion
		if (searchInput.length === 0) return true
```